test/pipelinefinally_test.go

```diff
@@ -347,29 +347,36 @@ spec:
 	// finaltaskconsumingdagtask1 has a reference to a task result from failed task
 	// finaltaskconsumingdagtask4 has a reference to a task result from skipped task with when expression
 	expectedSkippedTasks := []v1beta1.SkippedTask{{
-		Name: "dagtask3",
+		Name:   "dagtask3",
+		Reason: v1beta1.IsStoppingSkip,
 	}, {
-		Name: "dagtask4",
+		Name:   "dagtask4",
+		Reason: v1beta1.IsStoppingSkip,
 		WhenExpressions: v1beta1.WhenExpressions{{
 			Input:    "foo",
 			Operator: "notin",
 			Values:   []string{"foo"},
```

Comment:
this is an example scenario where the reason the `pipelinetask` is skipped is not its `when` expressions evaluating to false, but the `when` expressions are included - this is something we can address in a follow up pull request as discussed in https://github.com/tektoncd/pipeline/pull/4800#pullrequestreview-959569116